### PR TITLE
CSHARP-4182: Support for Range Indexes.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: bindings/cs/MongoDB.Libmongocrypt/CryptClient.cs [hunk 1]
```diff
@@ -105,11 +105,7 @@ public CryptContext StartEncryptionContext(string db, byte[] command)
         /// <summary>
         /// Starts an explicit encryption context.
         /// </summary>
-        /// <param name="key">The key id.</param>
-        /// <param name="encryptionAlgorithm">The encryption algorithm.</param>
-        /// <param name="message">The BSON message.</param>
-        /// <returns>A encryption context. </returns>
-        public CryptContext StartExplicitEncryptionContext(byte[] keyId, byte[] keyAltName, string queryType, long? contentionFactor, string encryptionAlgorithm, byte[] message)
+        public CryptContext StartExplicitEncryptionContext(byte[] keyId, byte[] keyAltName, string queryType, long? contentionFactor, string encryptionAlgorithm, byte[] message, byte[] rangeOptions, bool expressionMode = false)
```
Comment (BorisDog) [inline bindings/cs/MongoDB.Libmongocrypt/CryptClient.cs:108]: `isExpressionMode`?

File: bindings/cs/MongoDB.Libmongocrypt/Library.cs [hunk 2]
```diff
@@ -562,8 +572,12 @@ internal class Delegates
             public delegate bool mongocrypt_ctx_decrypt_init(ContextSafeHandle handle, BinarySafeHandle binary);
 
             [return: MarshalAs(UnmanagedType.I1)]
-            public delegate bool
-                mongocrypt_ctx_explicit_encrypt_init(ContextSafeHandle handle, BinarySafeHandle binary);
+            public delegate bool mongocrypt_ctx_explicit_encrypt_init(ContextSafeHandle handle, BinarySafeHandle binary);
+            /// <summary>
```
Comment (BorisDog) [inline bindings/cs/MongoDB.Libmongocrypt/Library.cs:576]: blank line before summary?

Does `mongocrypt_ctx_explicit_encrypt_init ` mentioned in summary refer to `mongocrypt_ctx_explicit_encrypt_expression_init` ?

File: bindings/cs/MongoDB.Libmongocrypt/CryptClient.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply bindings/cs/MongoDB.Libmongocrypt/CryptClient.cs:108]: done

File: bindings/cs/MongoDB.Libmongocrypt/Library.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply bindings/cs/MongoDB.Libmongocrypt/Library.cs:576]: good point, fixed

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**